moderation/punishments.go

```diff
@@ -217,11 +217,6 @@ func sendPunishDM(config *Config, dmMsg string, action ModlogAction, gs *dstate.
 }
 
 func KickUser(config *Config, guildID int64, channel *dstate.ChannelState, message *discordgo.Message, author *discordgo.User, reason string, user *discordgo.User, del int) error {
-	defer func() {
-		if r := recover(); r != nil {
-			logger.Infof("Recovered from panic: %#v", r)
-		}
-	}()
```

Comment:
This `recover` is completely safe to remove given the other patches in this PR. `KickUser` is called from three locations in the source:
* once in the kick command (commands.go:313)
  * We already recover from panics in command executions.
* once in automod legacy (automod_legacy/bot.go:167)
  * The necessary panic handler is added in this PR. 
* and once in advanced automod via the kick user effect (automod/effects.go:272).
  * The necessary panic handler is added is added in this PR.